useradmin/user-lib.pl

```diff
@@ -2724,4 +2724,20 @@ sub generate_random_password
 return $rv;
 }
 
+# config_pre_load(mod-info-refdd, [mod-order-ref])
+# Check if some config options are conditional,
+# and if not allowed, remove them from listing
+sub config_pre_load
+{
+my ($modconf_info, $modconf_order) = @_;
+my $unix_crypt_supports_yescrypt =
+		unix_crypt_supports_yescrypt();
+
+# Tweak available formats on the config if yescrypt
+# not available and cannot be used on given system
+if (!$unix_crypt_supports_yescrypt) {
+	$modconf_info->{'md5'} =~ s/,5-<tt>yescrypt<\/tt>//;
```

Comment:
Wouldn't it instead be better to show a warning on the module's main page if the selected hash format isn't supported? That would be less fragile ..

Reply:
> Wouldn't it instead be better to show a warning on the module's main page if the selected hash format isn't supported?

Yes, this is what was happening already.

> That would be less fragile

This is already quite solid it terms of the match. It cannot fail really (at least I don't see how it could) but I can easily remove it. I just thought that displaying an option that cannot be used is pointless.

So, do you really want me to remove it?

Reply:
Yeah, editing the hash from config.info like this just seems a little sketchy to me. And also it doesn't prevent selection of the wrong option via other means, like editing `/etc/webmin/useradmin/config` directly.

Reply:
Alright, reverted.